gcs/constraints/regular.hh

```diff
@@ -22,16 +22,21 @@ namespace gcs
         const std::vector<IntegerVariableID> _vars;
         const std::vector<Integer> _symbols;
         const long _num_states;
-        const std::vector<std::vector<long>> _transitions;
+        std::vector<std::unordered_map<Integer, long>> _transitions;
         const std::vector<long> _final_states;
 
     public:
         explicit Regular(std::vector<IntegerVariableID> vars,
             std::vector<Integer> symbols,
             long num_states,
-            std::vector<std::vector<long>> transitions,
+            std::vector<std::unordered_map<Integer, long>> transitions,
             std::vector<long> final_states);
 
+        Regular(std::vector<IntegerVariableID> v,
```

Comment:
`explicit`? Doesn't really matter for multi-argument constructors but I try to get into the habit of using `explicit` by default unless I have a good reason not to.